File: app/src/main/java/com/ignis/android_cleanarchitecture/presentation/di/module/DataModule.java
package com.ignis.android_cleanarchitecture.presentation.di.module;

import android.content.Context;

import com.ignis.android_cleanarchitecture.data.repository.AboutRepositoryImpl;
import com.ignis.android_cleanarchitecture.data.repository.CityRepositoryImpl;
import com.ignis.android_cleanarchitecture.data.repository.WeatherRepositoryImpl;
import com.ignis.android_cleanarchitecture.domain.repository.AboutRepository;
import com.ignis.android_cleanarchitecture.domain.repository.CityRepository;
import com.ignis.android_cleanarchitecture.domain.repository.WeatherRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DataModule {

    @Provides
    @Singleton
    public CityRepository provideCityRepository(Context context) {
        return new CityRepositoryImpl(context);
    }

    @Provides
    @Singleton
    public WeatherRepository provideWeatherRepository(Context context) {
        return new WeatherRepositoryImpl(context);
    }

    @Provides
    @Singleton
    public AboutRepository provideAboutRepository(Context context) {
        return new AboutRepositoryImpl(context);
    }

}

File: data/repository/src/main/java/com/kazakago/cleanarchitecture/data/repository/di/DataModule.kt
package com.kazakago.cleanarchitecture.data.repository.di

import com.kazakago.cleanarchitecture.data.repository.hierarchy.about.AboutRepositoryImpl
import com.kazakago.cleanarchitecture.data.repository.hierarchy.city.CityRepositoryImpl
import com.kazakago.cleanarchitecture.data.repository.hierarchy.weather.WeatherRepositoryImpl
import com.kazakago.cleanarchitecture.domain.repository.hierarchy.about.AboutRepository
import com.kazakago.cleanarchitecture.domain.repository.hierarchy.city.CityRepository
import com.kazakago.cleanarchitecture.domain.repository.hierarchy.weather.WeatherRepository
import org.koin.dsl.module

val dataModule = module {
    single<CityRepository> { CityRepositoryImpl(get()) }
    single<AboutRepository> { AboutRepositoryImpl(get()) }
    single<WeatherRepository> { WeatherRepositoryImpl(get()) }
}